### PR TITLE
Fix Nemotron 3 SFT example: correct LoRA target modules for hybrid Mamba architecture

### LINKED ISSUE #6773: Nemotron 3 SFT example uses target_modules that may not exist in hybrid architecture
## Problem

TRL's official Nemotron 3 SFT example (`examples/scripts/sft_nemotron_3.py`) uses LoRA target modules that may not exist in NemotronH's hybrid architecture:

```python
target_modules=["q_proj", "k_proj", "v_proj", "o_proj", 
                "gate_proj", "up_proj", "down_proj"]
```

The `gate_proj`, `up_proj`, `down_proj` modules are standard MLP projection names from Llama-style architectures. NemotronH uses Mamba-2 layers (with `in_proj`, `out_proj`) and MoE layers (with different naming). These MLP module names may silently not match anything in the model.

## Impact

Users following the official example may get LoRA adapters that only target a small fraction of the model's layers (the attention projections), with the MLP/Mamba modules silently skipped. This produces the same "loss not improving" failure mode we experienced:

- DPO LoRA training: 25 steps, loss stuck at 0.6931 (ln(2))
- Zero behavioral change in the model
- No warning that target modules didn't match

## Reproduction

We ran DPO LoRA on `nvidia/NVIDIA-Nemotron-Nano-9B-v2` with attention-only target modules (same as the example minus the MLP modules). The LoRA adapter only touched 4 out of 56 layers (7.1%). Loss never moved from ln(2). Full writeup: https://github.com/davidnichols-ops/claude-yolo-vibes-v5/blob/master/EVALUATION.md

## Proposed Fix

1. Update the example to use target modules that actually exist in NemotronH:

```python
# Inspect the model to find valid target modules
import re
target_modules = list(set(
    name.split(".")[-1] for name, module in model.named_modules()
    if isinstance(module, torch.nn.Linear) and 
    re.match(r".*\d+\..*", name)  # only layer-level modules
))
```

Or explicitly list NemotronH-appropriate modules:

```python
target_modules=[
    # Attention
    "q_proj", "k_proj", "v_proj", "o_proj",
    # Mamba (if PEFT allows — see PEFT issue about mamba_model_types ban)
    "in_proj", "out_proj", 
    # MoE/MLP — check actual module names in the model
    # "gate", "up", "down"  # verify these exist before including
]
```

2. Add a comment in the example explaining that NemotronH is a hybrid architecture and users should verify target modules with `model.named_modules()`.

3. Consider adding a runtime check in TRL that warns when configured `target_modules` don't match any modules in the model.

## Environment

- trl: 1.10.0
- transformers: 5.16.0.dev0
- Model: nvidia/NVIDIA-Nemotron-Nano-9B-v2

## Related

- PEFT PR #3289: Added default target_modules for nemotron_h (attention only)
- PEFT issue (filed separately): NemotronH LoRA target modules silent failure
- Full writeup: https://github.com/davidnichols-ops/claude-yolo-vibes-v5/blob/master/EVALUATION.md

### PR DESCRIPTION
**Pull request by davidnichols-ops**
## What does this PR do?

Fixes the LoRA `--lora_target_modules` in the Nemotron 3 SFT example (`examples/scripts/sft_nemotron_3.py`), which had two problems for the hybrid Mamba-Attention architecture:

1. **`gate_proj` does not exist in NemotronH.** The MLP layers use a non-gated `up_proj`/`down_proj` architecture (verified in `transformers/models/nemotron_h/modeling_nemotron_h.py` — `NemotronHMLP` has only `up_proj`/`down_proj`, and `NemotronHExperts` explicitly comments "No gating mechanism unlike Mixtral"). PEFT silently skips target module names that do not match any module in the model, so `gate_proj` was a no-op.

2. **Mamba layers were missing from the target list.** NemotronH is a hybrid Mamba-Attention architecture where the majority of layers are Mamba SSM blocks with `in_proj`/`out_proj` projections. The previous target list only covered attention and MLP layers, leaving the Mamba layers frozen during fine-tuning. This PR adds `in_proj`, the Mamba input projection, which is a standard `nn.Linear` called as a module in the forward pass and is not blocked by PEFT's Mamba compatibility check.

`out_proj` is intentionally **not** added. PEFT raises `ValueError` when `out_proj` is targeted on `nemotron_h` (and other Mamba-based model types) because the fused SSM kernel (`mamba_split_conv1d_scan_combined`) consumes `out_proj.weight` as a raw tensor argument, bypassing the LoRA-wrapped module's `forward()` method. This means LoRA on `out_proj` would be a silent no-op during training and a shape mismatch crash at `merge_and_unload()`. The ban is enforced in `peft/tuners/tuners_utils.py` (`_check_lora_target_modules_mamba`, `incompatible_modules = {"out_proj", "conv1d"}`).

Fixes #6773

## Before submitting

- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request), Pull Request section?
- [x] Was this discussed/approved via a GitHub issue? Please add a link to it if that's the case. — #6773
- [x] Did you make sure to update the documentation with your changes? — Added explanatory comment in the docstring describing which modules are targeted and why `gate_proj`/`out_proj` are excluded.
- [x] Did you write any new necessary tests? — N/A, this is a one-line fix to an example script docstring. No testable code path changed.

## AI writing disclosure

We welcome the use of AI tools to help with contributions. For transparency and to help us improve our review process, please indicate the level of AI involvement in this PR.

- [ ] No AI usage: the PR was written entirely by a human.
- [x] AI-assisted: some parts were suggested or improved by AI, but the PR was written and reviewed by a human. The technical analysis (verifying module names against `modeling_nemotron_h.py`, checking PEFT's `_check_lora_target_modules_mamba` ban list, confirming `in_proj` is called as a module vs `out_proj` passed as a raw tensor) was performed by reviewing the source code directly.
- [ ] AI-generated: the PR was mostly or fully generated by an AI tool.

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag members/contributors who may be interested in your PR. @qgallouedec — this supersedes the closed #6777. That PR added `out_proj` to the target list, which PEFT rejects with `ValueError` for `nemotron_h`. This PR adds only `in_proj` (not banned, called as a module) and removes `gate_proj` (does not exist).

### CODE REVIEW

**Comment by davidnichols-ops** [comment]
Closing PR — too much work for HF engineers.
